From bdb2e7ec22d9030c741f7da5b794fde3b778a956 Mon Sep 17 00:00:00 2001
From: ghostop14 <ghostop14@gmail.com>
Date: Sun, 29 Mar 2020 18:56:53 -0400
Subject: [PATCH] Fix test_xtrx_soapy build issue

The Soapy setupstream function now passes back rxStream as the
return value rather than taking it as a parameter.  This PR
corrects the code to build correctly and addresses issue #23.
---
 soapy/test_xtrx_soapy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/soapy/test_xtrx_soapy.c b/soapy/test_xtrx_soapy.c
index de1c52c..7221410 100644
--- a/soapy/test_xtrx_soapy.c
+++ b/soapy/test_xtrx_soapy.c
@@ -67,8 +67,8 @@ int main(void)
 
     //setup a stream (complex floats)
     SoapySDRStream *rxStream;
-    if (SoapySDRDevice_setupStream(sdr, &rxStream, SOAPY_SDR_RX, SOAPY_SDR_CF32, NULL, 0, NULL) != 0)
-    {
+    rxStream = SoapySDRDevice_setupStream(sdr, SOAPY_SDR_RX, SOAPY_SDR_CF32, NULL, 0, NULL);
+    if ( rxStream == 0)    {
         printf("setupStream fail: %s\n", SoapySDRDevice_lastError());
     }
     SoapySDRDevice_activateStream(sdr, rxStream, 0, 0, 0); //start streaming
-- 
2.26.2

